File: 3300. minimum-element-after-replacement-with-digit-sum.py
class Solution:
    def minElement(self, nums: list[int]) -> int:
        min_val = float('inf')
        
        for num in nums:
            current_sum = 0
            
            while num > 0:
                current_sum += num % 10
                num //= 10
            
            min_val = min(min_val, current_sum)
                
        return min_val

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1
    nums1 = [10,12,13,14]
    output1 = sol.minElement(nums1)
    print(f"Example 1: nums={nums1}")
    print(f"Output: {output1}")
    print(f"Expected: 1")
    print()
    
    # Example 2
    nums2 = [1,2,3,4]
    output2 = sol.minElement(nums2)
    print(f"Example 2: nums={nums2}")
    print(f"Output: {output2}")
    print(f"Expected: 1")
    print()
    
    # Example 3
    nums3 = [999,19,199]
    output3 = sol.minElement(nums3)
    print(f"Example 3: nums={nums3}")
    print(f"Output: {output3}")
    print(f"Expected: 10")

File: 3300. minimum-element-after-replacement-with-digit-sum.java
class Solution {
    public int minElement(int[] nums) {
        int minVal = Integer.MAX_VALUE;
        
        for (int num : nums) {
            int currentSum = 0;
            
            while (num > 0) {
                currentSum += num % 10;
                num /= 10;
            }
            
            minVal = Math.min(minVal, currentSum);
        }
        
        return minVal;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1
        int[] nums1 = {10,12,13,14};
        int output1 = sol.minElement(nums1);
        System.out.println("Example 1: nums=[10,12,13,14]");
        System.out.println("Output: " + output1);
        System.out.println("Expected: 1");
        System.out.println();
        
        // Example 2
        int[] nums2 = {1,2,3,4};
        int output2 = sol.minElement(nums2);
        System.out.println("Example 2: nums=[1,2,3,4]");
        System.out.println("Output: " + output2);
        System.out.println("Expected: 1");
        System.out.println();
        
        // Example 3
        int[] nums3 = {999,19,199};
        int output3 = sol.minElement(nums3);
        System.out.println("Example 3: nums=[999,19,199]");
        System.out.println("Output: " + output3);
        System.out.println("Expected: 10");
    }
}